include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -105,25 +77,129 @@ template <typename _ExecutionPolicy>
 using __allow_parallel = typename __internal::__policy_traits<::std::decay_t<_ExecutionPolicy>>::__allow_parallel;
 
 template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_vectorization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_vector(),
-                                       __internal::__is_random_access_iterator_t<_IteratorTypes...>()))
+constexpr auto
+__is_vectorization_preferred()
 {
-    return __internal::__lazy_and(__exec.__allow_vector(),
-                                  __internal::__is_random_access_iterator_t<_IteratorTypes...>());
+    return ::std::conjunction<decltype(::std::decay_t<_ExecutionPolicy>::__allow_vector()),
+                              __internal::__is_random_access_iterator_t<_IteratorTypes...>>();
 }
 
 template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_parallelization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_parallel(),
-                                       __internal::__is_random_access_iterator_t<_IteratorTypes...>()))
+constexpr auto
+__is_parallelization_preferred()
+{
+    return ::std::conjunction<decltype(::std::decay_t<_ExecutionPolicy>::__allow_parallel()),
+                              __internal::__is_random_access_iterator_t<_IteratorTypes...>>();
+}
+
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+struct __tbb_backend_tag
+{
+};
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+};
+
+template <class _IsVector>
+struct __parallel_tag
+{
+    using __is_vector = _IsVector;
+    // backend tag can be change depending on
+    // TBB availability in the environment
+    using __backend_tag = __tbb_backend_tag;
```

Comment:
I guess a backend type should be depended on the environment and a logic of backend selection should be like that we have currently:

```
namespace oneapi
{
namespace dpl
{
#if _ONEDPL_PAR_BACKEND_TBB
namespace __par_backend = __tbb_backend;
#elif _ONEDPL_PAR_BACKEND_OPENMP
namespace __par_backend = __omp_backend;
#elif _ONEDPL_PAR_BACKEND_SERIAL
namespace __par_backend = __serial_backend;
#else
#    error "Parallel backend was not specified"
#endif
} // namespace dpl
} // namespace oneapi
```
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/parallel_backend.h#L43

Reply:
Fixed